docs/main-concepts/how-to-define-a-fetch.md

```diff
@@ -0,0 +1,352 @@
+
+The suggested naming convention is: **{[CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) operation} {Resource name}**
+
+e.g.
+
+- `Get Person`
+- `Get People`
+- `Update Config`
+- `Create Book`
+
+> **NOTE**: it's not recommended to include the word "fetch" in your display name because it's redundant in the dev tools.
+
+---
+
+## The `make` function and the `key`
+
+The `make` function defines the functionality of a fetch factory.
+
+Depending on your `key`, there are two types of fetch factories you can make:
+
+1. A fetch factory that produces many unique fetches
+2. A fetch factory that can only produce one unique fetch
+
+### Defining a fetch factory that produces many different fetches
+
+As a refresher, a fetch factory produces a fetch like so:
+
+```js
+const personFetch = makePersonFetch('person-id-123');
+```
+
+And, depending on the fetch factory, you may be able to produce more than one unique fetch from the same fetch factory e.g.
+
+```js
+function Example() {
+  const matFetch = makePersonFetch('person-id-mat');
+  const pearlFetch = makePersonFetch('person-id-pearl');
+
+  const [mat] = useFetch(matFetch);
+  const [pearl] = useFetch(pearlFetch);
+
+  if (!mat || !pearl) return null;
+
+  return (
+    <div>
+      Hello, <div>{mat.name}</div> and <div>{pearl.name}</div>!
+    </div>
+  );
+}
+```
+
+The number of unique fetches a fetch factory can produce is determined by the number of unique `key` possibilities.
+
+#### What's a `key`?
+
+The **`key`** is an array of strings (or numbers) that will be used to store and lookup the data associated with your fetch.
+
+The `key` is defined in the object returned by `make`:
+
+```js
+const makePersonFetch = defineFetch({
+  displayName: 'Get Person',
+  make: personId => ({
+    // this 👇👇👇 is the "key" of this fetch
+    key: [personId],
+    //     👆👆👆
+
+    request: // ...
+  }),
+});
+```
+
+In the `Example` component above, when different `personId`s were passed into the fetch factory, they returned different fetch instances. However if you pass the same `personId` into the fetch factory, you'd get back the same fetch instance.
+
+```js
+const matFetch_1 = makePersonFetch('person-id-mat');
+const matFetch_2 = makePersonFetch('person-id-mat');
+
+console.log(matFetch_1 === matFetch_2); // true
+```
+
+> **Technical Note:** This works because the fetch factory is [_memoized_](https://en.wikipedia.org/wiki/Memoization) against the key to return the same fetch reference.
+
+### Defining a fetch factory that can only produce one unique fetch
+
+On the flip side, if your `key` is defined as an empty array, then it can only produce one unique fetch instance because, no matter the fetch factory arguments, there is only one possible key—the empty array.
```

Comment:
This is more of a curiosity for myself. Could you also just accomplish this by using a static string inside of the array? A concept of having no `key` is weird to me. 

Also, couldn't you just make it so if you give a `null` or `undefined` key it would have this behavior instead of passing an empty array? Empty array seems a bit odd.

(This isn't really doc feedback, not suggesting you change anything right now)

Reply:
I was thinking about allowing the user to pass `null` but passing an empty array covered this case well. This is what [React did with `useEffect`](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects) and its dependencies array.

> If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. This tells React that your effect doesn’t depend on any values from props or state, so it never needs to re-run. **This isn’t handled as a special case — it follows directly from how the dependencies array always works.**

[source](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects)